trade-finance-manager-ui/server/controllers/case/activity/helpers/map-activities.ts

```diff
@@ -0,0 +1,50 @@
+import { ACTIVITY_TYPES, TfmActivity } from '@ukef/dtfs2-common';
+import { format, fromUnixTime } from 'date-fns';
+
+export const mapActivities = (activities: TfmActivity[] | undefined) => {
+  if (!activities) {
+    return [];
+  }
+
+  return activities.map((activity) => {
+    const datetime = {
+      timestamp: fromUnixTime(activity.timestamp),
+      type: 'datetime',
+    };
+    const byline = {
+      text: `${activity.author.firstName} ${activity.author.lastName}`,
+    };
+
+    switch (activity.type) {
+      case ACTIVITY_TYPES.CANCELLATION:
+        return {
+          label: {
+            text: 'Deal cancelled',
+          },
+          html: `
+          <p> Deal stage:
```

Comment:
I don't really like having the html generated here but Im not sure how else to effectively do this